rcode-reader/src/main/java/eu/rssw/pct/elements/TableElement.java

```diff
@@ -8,52 +27,69 @@
 import eu.rssw.pct.RCodeInfo;
 
 public class TableElement extends AbstractAccessibleElement {
-  protected int flags;
-  protected int crc;
-  protected int prvte;
-  protected VariableElement[] fields;
-  protected IndexElement[] indexes;
+  private final int flags;
+  private final int prvte;
+  private final VariableElement[] fields;
+  private final IndexElement[] indexes;
   private String beforeTableName;
 
-  public TableElement(String name, Set<AccessType> accessType, byte[] segment, int currentPos, int textAreaOffset,
-      ByteOrder order) {
+  public TableElement(String name, Set<AccessType> accessType, int flags, int prvte, VariableElement[] fields,
+      IndexElement[] indexes, String beforeTableName) {
     super(name, accessType);
+    this.fields = fields;
+    this.indexes = indexes;
+    this.beforeTableName = beforeTableName;
+    this.flags = flags;
+    this.prvte = prvte;
+  }
 
+  public static TableElement fromDebugSegment(String name, Set<AccessType> accessType, byte[] segment, int currentPos,
+      int textAreaOffset, ByteOrder order) {
     int fieldCount = ByteBuffer.wrap(segment, currentPos, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
     int indexCount = ByteBuffer.wrap(segment, currentPos + 2, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
-    this.flags = ByteBuffer.wrap(segment, currentPos + 4, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
-    this.crc = ByteBuffer.wrap(segment, currentPos + 6, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
-    this.prvte = ByteBuffer.wrap(segment, currentPos + 8, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
+    int flags = ByteBuffer.wrap(segment, currentPos + 4, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
+    // int crc = ByteBuffer.wrap(segment, currentPos + 6, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ACommentedOutCodeLine)